src/git/tests/test_server.py

```diff
@@ -0,0 +1,24 @@
+import pytest
+from pathlib import Path
+import git
+from mcp_server_git.server import git_checkout
+
+def test_git_checkout_existing_branch(tmp_path: Path):
+    # Setup test repo
+    repo = git.Repo.init(tmp_path)
+    Path(tmp_path / "test.txt").write_text("test")
+    repo.index.add(["test.txt"])
+    repo.index.commit("initial commit")
+
+    # Create and test branch
+    repo.git.branch("test-branch")
+    result = git_checkout(repo, "test-branch")
+
+    assert "Switched to branch 'test-branch'" in result
+    assert repo.active_branch.name == "test-branch"
+
+def test_git_checkout_nonexistent_branch(tmp_path: Path):
+    repo = git.Repo.init(tmp_path)
+
+    with pytest.raises(git.GitCommandError):
+        git_checkout(repo, "nonexistent-branch")
```

Comment:
I am not sure we should add this test. It feels straight forward enough.

If we do add tests, we should use a proper fixture:
```python
@pytest.fixture
def test_repository():
    # Setup
    temp_path = Path("temp_test_repo")
    temp_path.mkdir(exist_ok=True)
    
    # Initialize your repository here
    # ...
    
    yield temp_path
    
    # Cleanup
    shutil.rmtree(temp_path)

def test_git_checkout_existing_branch(test_repository):
   ...
```

Reply:
@dsp-ant I think I've addressed your concern? Let me know what you think.